frontend/benefit/applicant/src/components/applications/forms/application/step1/ApplicationFormStep1.tsx

```diff
@@ -149,19 +150,33 @@ const ApplicationFormStep1: React.FC<DynamicFormStepComponentProps> = ({
           </StyledFormGroup>
         )}
         <StyledFormGroup>
-          <TextInput
-            id={fields.companyIban.name}
-            name={fields.companyIban.name}
-            label={fields.companyIban.label}
-            placeholder={fields.companyIban.placeholder}
-            onChange={formik.handleChange}
-            onBlur={formik.handleBlur}
+          <InputMask
+            mask={fields.companyIban.mask?.format ?? ''}
+            maskChar={null}
             value={formik.values.companyIban}
-            invalid={!!getErrorMessage(APPLICATION_FIELDS.COMPANY_IBAN)}
-            aria-invalid={!!getErrorMessage(APPLICATION_FIELDS.COMPANY_IBAN)}
-            errorText={getErrorMessage(APPLICATION_FIELDS.COMPANY_IBAN)}
-            required
-          />
+            onBlur={formik.handleBlur}
+            onChange={(e) => {
+              const initValue = e.target.value;
+              const value =
+                fields.companyIban.mask?.stripVal(initValue) ?? initValue;
+              return formik.setFieldValue(fields.companyIban.name, value);
```

Comment:
is setting field value enough? Not sure how formik works under the hood, but perhaps 
```
              e.target.value = value;
              return formik.handleChange(e);
```
could be safer?

Reply:
I followed some of the Formik members' recommendations in this discussion, like this [one](https://github.com/formium/formik/issues/1243#issuecomment-461176174).

Reply:
👍